packages/jellyfish-transaction-builder/__tests__/txn/txn_builder_submit_ext_htlc.test.ts

```diff
@@ -0,0 +1,208 @@
+import { MasterNodeRegTestContainer, GenesisKeys } from '@defichain/testcontainers'
+import { getProviders, MockProviders } from '../provider.mock'
+import { P2WPKHTransactionBuilder } from '../../src'
+import { calculateTxid, sendTransaction } from '../test.utils'
+import { WIF } from '@defichain/jellyfish-crypto'
+import BigNumber from 'bignumber.js'
+import { OP_CODES } from '@defichain/jellyfish-transaction'
+import { ICXSubmitEXTHTLC } from '@defichain/jellyfish-transaction/script/dftx/dftx_icxorderbook'
+import { ICXEXTHTLCInfo, ICXListHTLCOptions } from '@defichain/jellyfish-api-core/category/icxorderbook'
+import { Testing } from '@defichain/jellyfish-testing'
+import { icxorderbook } from '@defichain/jellyfish-api-core'
+
+describe('submit EXT HTLC', () => {
+  const container = new MasterNodeRegTestContainer()
+  const testing = Testing.create(container)
+  let providers: MockProviders
+  let builder: P2WPKHTransactionBuilder
+
+  beforeAll(async () => {
+    await testing.container.start()
+    await testing.container.waitForWalletCoinbaseMaturity()
+
+    providers = await getProviders(container)
+    providers.setEllipticPair(WIF.asEllipticPair(GenesisKeys[0].owner.privKey)) // set it to container default
+    builder = new P2WPKHTransactionBuilder(providers.fee, providers.prevout, providers.elliptic)
+
+    // steps required for ICX testing
+    await testing.icxorderbook.setAccounts(await providers.getAddress(), await providers.getAddress())
+    await testing.rpc.account.utxosToAccount({ [testing.icxorderbook.accountDFI]: `${500}@${testing.icxorderbook.symbolDFI}` })
+    await testing.rpc.account.utxosToAccount({ [testing.icxorderbook.accountBTC]: `${10}@${testing.icxorderbook.symbolDFI}` }) // for fee
+    await testing.generate(1)
+    await testing.fixture.createPoolPair({
+      a: { amount: '1', symbol: testing.icxorderbook.symbolBTC },
+      b: { amount: '100', symbol: testing.icxorderbook.symbolDFI }
+    })
+    testing.icxorderbook.DEX_DFI_PER_BTC_RATE = new BigNumber(100 / 1)
```

Comment:
new BigNumber(100)

Reply:
new BigNumber(100 / 1) is actually easier to understand in this case

Reply:
or else
```suggestion
    testing.icxorderbook.DEX_DFI_PER_BTC_RATE = new BigNumber(100) // 100 / 1
```

Reply:
yes, the purpose is to give some additional details on how we get `DEX_DFI_PER_BTC_RATE `